eclipse-theia/theia
# Enable user-level task configurations

Signed-off-by: Thomas Mäder <tmader@redhat.com>
Fixes: #7446 (Support User Level Tasks)


#### What it does
This PR adds support for adding tasks in ~/.theia/tasks.json. It does so by extending the support for preference configurations to user-level preferences and accessing that preference scope directly for reading tasks. That code has been refactored a bit to enable this.
Contributed tasks can not be configured as user-level tasks, and only shell and terminal task types work as user-level tasks (same as in VS Code.
A command to "Tasks: Open User Tasks" has been added that opens an editor on the user-level tasks.json

#### How to test
* Go through the various commands starting with "Tasks:..." and make sure they still function properly for folder-level tasks.
* Add tasks to ~/.theia/tasks.json and verify they work as expected.

#### Known problems
* Currently, the combined schema for all task types is used in the user tasks.json, whereas only the shell and terminal types should be allowed in that file. I would postpone that fix to keep this PR more manageable.

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
